Server/src/services/db/tracker.controller.ts

```diff
@@ -47,7 +48,7 @@ export default class TrackerController {
 	 * @param data - New additional data field (Optional)
 	 * @returns Tracker | null if an error occurs
 	 */
-	public async update(uid: string, vehicleId?: number, data?: any): Promise<Tracker | null> {
+	public async update(uid: string, vehicleId: number | null, data?: any): Promise<Tracker | null> {
```

Comment:
This will probably be fine in our use case, but now there is no way to only update the `data` of a tracker. I'm not sure if there is something interesting there, and if we might to update it regardless of the associated vehicle, so I would suggest still allowing `undefined` for the `vehicleId` here. (Yes, I know we will get into the semantic challenges regarding `null` vs `undefined` here...)

Reply:
Another idea is to change the order of the arguments. That means we can use data before vehicleId which would allow us to still change data but keep the semantics in the same way.

Reply:
Correction: This doesn't work, as a required parameter cannot follow an optional parameter